src/components/People.js

```diff
@@ -0,0 +1,64 @@
+import { Component } from "react";
+import "../App.css";
+
+class People extends Component{
+
+    constructor(){
+        super();
+        this.state={
+            people:[],
+            searchInput:"",
+        }
+    }
+  
+    fetchPeople=()=>{
+        fetch("https://ghibliapi.herokuapp.com/people")
+        .then(res=>res.json())
+        .then ((data)=>{
+            this.setState({
+                people: data,
+            })
+        })
+    }
+
+    componentDidMount(){
+        this.fetchPeople();
+    }
+
+    handleInput=(e)=>{
+        e.preventDefault();
+        this.setState({
+            searchInput: e.target.value
+        })
+    }
+
+    // handleSearch=(e)=>{
+    //     e.preventDefault(); 
+    // }
+
```

Comment:
```suggestion
     handleSearch=(e)=>{
         e.preventDefault(); 
         let peopleList = this.state.people.map((person)=>{
          if(person.name === this.state.searchInput){
            return(
                <div>
                    <div>Name:{person.name}</div>
                    <div>Age:{person.age}</div>
                    <div>Gender:{person.gender}</div>
                </div>
            )
          }
      })
         this.setState({
         selectedCharacter: peopleList,
         searchInput: "" //<-- clear out searchInput to reset the form
         })
     }

```
Here's how you can modify your existing code to make it so that you update who the `selectedPerson` is whenever the user makes a new search.

Try using React developer tools to see what `this.state.selectedPerson` becomes when you make a search.

Your way of using `.map` is just one way of making this work, but I would actually recommend using `.find` in this situation instead of `.map`. If there's time during flex-time tomorrow, we can go over this some more.